portal-impl/src/com/liferay/portal/util/PortalClassPathUtil.java

```diff
@@ -107,23 +111,28 @@ public static void initializeClassPaths(ServletContext servletContext) {
 			classLoader = currentThread.getContextClassLoader();
 		}
 
-		StringBundler sb = new StringBundler(8);
+		List<File> fileList = _listClassPathFiles(
+			classLoader, ServletException.class.getName(),
+			CentralizedThreadLocal.class.getName(),
+			"com.liferay.shielded.container.ShieldedContainerInitializer");
 
-		sb.append(
-			_buildClassPath(classLoader, ServletException.class.getName()));
+		StringBundler sb = new StringBundler();
+		StringBundler bootstrapSB = new StringBundler();
 
-		sb.append(File.pathSeparator);
-		sb.append(
-			_buildClassPath(
-				classLoader, CentralizedThreadLocal.class.getName()));
+		if (fileList != null) {
```

Comment:
@julschong When fileList is null which is possible based on your logic in _listClassPathFiles, you should not move on to the initialize logic, as bootstrapSB will be empty for sure which is not right. The bootstrap class path is used to start the new process, it can not be empty.